### ISSUE TITLE
AWS Cloud Infrastructure

### ISSUE THREAD

**Issue by chriswilty** [ScottLogic/prompt-injection#920]
This issue represents all others in the [Remote Deployment milestone](https://github.com/ScottLogic/prompt-injection/milestone/4?closed=1) already merged into the cloud infrastructure feature branch.

The end result is a set of CDK-defined resource stacks, synthesized into CloudFormation templates, orchestrated by a CodePipeline to trigger redeployment on merges into `main` branch.

Once this is merged to dev, then subsequently into main, I will cut a minor release. Following that, the pipeline will be deployed to AWS so that subsequent merges into main will automatically trigger deployment of all updated resources.
